plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -1805,6 +1805,14 @@ protected void ensureDiskControllers(VirtualMachineMO vmMo, Pair<String, String>
         int requiredNumScsiControllers = VmwareHelper.MAX_SCSI_CONTROLLER_COUNT - scsiControllerInfo.first();
         int availableBusNum = scsiControllerInfo.second() + 1; // method returned current max. bus number
 
+        if (DiskControllerType.getType(scsiDiskController) != scsiControllerInfo.third()) {
+            s_logger.debug(String.format("Change controller type from: %s to: %s", scsiControllerInfo.third().toString(),
+                    scsiDiskController));
+            vmMo.tearDownDevices(new Class<?>[]{VirtualSCSIController.class});
+            vmMo.addScsiDeviceControllers(DiskControllerType.getType(scsiDiskController));
```

Comment:
Hi @nvazquez What in case the root and data disk scsi controllers are different? root disk scsi controller considered as the first preference? Any doc needed.

Also, does this address the case when root disk controller is ide and data disk scsi controller is updated?

Reply:
> Also, does this address the case when root disk controller is ide and data disk scsi controller is updated?

and vice versa

Reply:
Those cases are handled by the method `HypervisorHostHelper.getScsiController()` which sets the `scsiDiskController` value depending on the root and data disk controllers and the recommended controller.